src/6_branch.js

```diff
@@ -76,7 +80,6 @@ var wrap = function(parameters, func, init) {
 				else if (parameters === callback_params.CALLBACK_ERR_DATA) {
 					callback(err, data);
 				}
-				next();
```

Comment:
one more thing, if you resume this `next();` here, make sure it doesn't get called twice for the case where we have err and the parameters is not NO_CALLBACK.

This is, btw, why I originally recommended putting your logic in the same if statement as where the throw is located. But let me know if this doesn't make sense